feat: merge patch RVX/anddea

anddea opened the pull request:
https://github.com/inotia00/revanced-integrations/pull/43

1. I reordered the YouTube settings, moving General to the top and rearranging elements as they appear in the UI. Additionally, I moved similar preferences closer to each other. The icons in the settings make it much easier to find elements, as they are displayed in YouTube's UI. But this reordering changed your defaults, so feel free to change it.
2. Since I didn't have any time, I couldn't improve the search functionality as I wanted (for now). However, I left TODOs, so you or other contributors can continue this work. But still, search is functioning and is fine for everyone as it is right now.
3. Overall, I left comments and documentation in some places. I suggest you to do the same: leave comments and documentation so other contributors can continue after us. 

To inform about each feature, you can copy the commit messages. Feel free to change code/resources however you want, as I currently don't have the time. 

Some notes:
1. The `Force snackbar theme` does not change the snackbar for the player (e.g., quality change, captions, etc.). I couldn't find the responsible resource; it is probably implemented in the code (Jetpack Compose? I'm not sure, as I didn't have time to check).
2. The `Add splash animation` patch can be integrated into the `Custom branding icon` patch and is needed only for the MMT animation.


TODO:
- [x] Minimize changes in `dev` branch
- [x] Check if a resource is valid
- [x] Lint code
- [x] Clarify patch option

anddea committed:
feat(YouTube - Shorts components): Add `Hide disabled comments button` and `Hide live chat header`

anddea committed:
feat(Custom branding icon): Add more icons

anddea committed:
feat(YouTube): Add `Shorts overlay buttons` patch

Replaces "Shorts outline patch".

anddea committed:
feat(YouTube): Add `Add splash animation` patch

Needed for MMT (custom branding icon) animation. Although, it could be integrated inside "Custom branding icon" patch as a booleanPatchOption.

anddea committed:
feat(YouTube): Add `Visual preferences icons`

anddea committed:
feat(YouTube): Add `Force snackbar theme` patch

anddea committed:
feat(YouTube - Settings): Add options to change position and title of RVX settings

anddea committed:
feat(Translations): Add options to remove languages and provide own translation / strings.xml

anddea committed:
feat(YouTube): Add search bar in settings

anddea committed:
chore(YouTube - Translations): Update LANGUAGES array

anddea committed:
refactor(YouTube): Fix typos in strings

inotia00 committed:
fix(YouTube/Custom branding icon YouTube): inclueded by default

inotia00 commented:
Because there are so many changes, reviewing PRs can take a long time.

I will try to review as quickly as possible, but I recommend not adding new changes to the PR branch if possible.

inotia00 commented:
Because default values ​​are already in Integrations, default values ​​do not need to be included in the XML

inotia00 committed:
fix(YouTube/Settings): remove default values from XML

inotia00 commented:
Vanced Manager's license is GPL v3, but YouTube Vanced's license is not GPL v3
Even though YouTube Vanced has been discontinued, you cannot include YouTube Vanced's icon in the default icon preset

inotia00 committed:
feat(Custom branding icon): remove Vanced icon preset

anddea commented:
> I reordered the ~~YouTube~~ RVX settings

Also, for other localizations, it appears messy, so ordering alphabetically (for English) does not seem like a good idea after all.

inotia00 commented:
The refactored RVX doesn't adhere to lexicographic arrangement as perfectly as before, but it nonetheless tries to preserve lexicographic arrangement as much as possible
(Because most users still speak English)

The arrangement of settings in RVX follows the official ReVanced settings, and I recommend not altering this order as much as possible to keep track of the official ReVanced source

anddea commented:
> to keep track of the official ReVanced source

I don't think the keys (such as key, title, summary, etc.) are the same as in ReVanced. Are they?

If you revert to the original settings order but include settings icons, you'll notice the mess in some places. But of course, it's up to you. You can just revert that one `revanced-prefs.xml` file.

If you include icons, you'll navigate by them once you get used to them. However, if you see a different order than in the UI, it may cause confusion.

inotia00 commented:
> I don't think the keys (such as key, title, summary, etc.) are the same as in ReVanced. Are they?

In [revanced-patches-4.7.1](https://github.com/inotia00/revanced-patches/releases/tag/v4.7.1) I added a number of changes to match ReVanced's keys, titles, and summaries as much as possible.

> If you include icons, you'll navigate by them once you get used to them. However, if you see a different order than in the UI, it may cause confusion.

I think this still needs to be considered.
Honestly, I've never used the `Visual preferences icons` patch so I don't think I understand it.

I will think about it for another day and then make a decision.

YT-Advanced commented on src/main/kotlin/app/revanced/patches/youtube/shorts/overlaybuttons/ShortsOverlayButtonsPatch.kt:
```diff
@@ -0,0 +1,86 @@
+        values = mapOf(
+            "Outline" to "outline",
+            "OutlineCircle" to "outlinecircle",
+            DEFAULT_ICON_KEY to "tiktok"
```
Does using Tiktok icon violate their intellectual property rights?

anddea replied:
They are not exactly TikTok icons, they made look-a-like without touching original TikTok icons.

YT-Advanced commented on src/main/kotlin/app/revanced/patches/youtube/utils/settings/SettingsPatch.kt:
```diff
@@ -38,6 +42,49 @@ object SettingsPatch : BaseResourcePatch(
+        DEFAULT_ELEMENT to "@string/about_key"
+    )
+
+    private val InsertPosition by stringPatchOption(
```
This string options appear to be unused, isn't it ?

anddea replied:
Nope. Check line 188. It uses both key and value as a fallback in case different people provide either one of them.

YT-Advanced commented on src/main/kotlin/app/revanced/patches/youtube/utils/settings/SettingsPatch.kt:
```diff
@@ -38,6 +42,49 @@ object SettingsPatch : BaseResourcePatch(
+        description = "Specify the setting name before which the RVX setting should be inserted."
+    )
+
+    private val CustomName by stringPatchOption(
```
I'm not sure if having such a `Custom Extended name` feature is a good idea, as the number of apps masquerading as Vanced will probably increase. 

Moreover, is it worth sacrificing performance for such a small feature?

Francesco146 replied:
in my opinion it doesn't impact performance at all. on the fact that it can increase fictitious apps.. yeah, a bit. like all branding patches tho. it takes a second to change headers, icons and appicons etc

anddea replied:
Well, it's a customization option, why should we restrict people, right? 

Given that most of the people won't change the name, it will break out of "heavy" code algorithm before it starts. 

But let's define, what's "heavy" and why it won't really cause any performance issues. 

This code goes through each language file (L) and each `<string>` element (S) inside those language files, so in worst case scenario it will be O(L×S) which in the worst case can be considered as O(n•n) or O(n²) which is common for algorithms with nested loops and we have a lot of code with nested loops in almost all projects. You probably already know that it's somewhere in the middle of Big O. But take a note that it is the worst case scenario. 

What we have in reality? "revanced_extended_settings_title" goes somewhere in the beginning of each language file, so it's faster, for the best case scenario it will be O(L). Except some of the language files don't even have this string at all and it will pass the whole file, that's why it'll be O(L•S) and I hope you know that it's faster than O(n²) as L and S are not equal numbers. To improve it, we can manually provide those 2-3 language files that have this string. You won't notice any performance issues at all, it'll change the name in milliseconds even on slow devices.

anddea commented:
> needs to be considered

It's just a customization option that can be turned off like any other patch, it can be also integrated inside `Settings` patch as boolean patch option. 

Multiple people already opened issues in your repo asking for it and there are a lot of people coming to my repo for this feature that's why I opened it, so people don't use multiple forks. Of course, all those icons and resources take more space and increase patches size. And with new changes, this patch might need to be updated as well. 

I created a poll for this feature if it's helpful. 
https://github.com/anddea/revanced-patches/discussions/130

Edit: It's not a problem if you remove it, I just let you know about these changes and why they were done
> Feel free to change code/resources however you want

inotia00 commented:
I mean, I'm not saying I won't merge this PR

I am talking about a change in the arrangement order of settings

I reorganized the settings arrangement to match the order of the official ReVanced settings, but PR is reverting this change
(For example, this PR places the `General` PreferenceScreen first)

If possible I would like to keep the arrangement order of the settings current

Francesco146 commented:
he's talking about the `Visual preferences icons` patch, not the order of the settings which is best kept the same from ReVanced

anddea commented:
Ofc, you can revert it back by just copy-pasting your current arrangement. I've fixed some typos in keys, so only those have to be kept. 

https://github.com/inotia00/revanced-patches/pull/55/commits/3609314932e2fc5b527daa73fe517300abcea060

And added key for this preference (used for visual preferences icons)
https://github.com/inotia00/revanced-patches/blob/1558aacd81873fdc797cffd2aa48b23f410c3f27/src/main/resources/youtube/settings/xml/revanced_prefs.xml#L577

I think that's  it, don't remember if there are other changes for settings icons. 

Although, for me it seems better when General is on top. Also, there was a request for this before by someone.

KobeW50 commented on src/main/kotlin/app/revanced/patches/music/layout/branding/icon/CustomBrandingIconPatch.kt:
```diff
@@ -21,7 +22,17 @@ object CustomBrandingIconPatch : BaseResourcePatch(
+        key = "SplashHeaderIcon",
+        default = true,
+        title = "Splash and header icons",
+        description = "Apply custom branding icon to Splash and Header.",
```
```suggestion
        description = "Apply the custom branding icon to the splash screen and header.",
```

KobeW50 commented on src/main/kotlin/app/revanced/patches/youtube/general/snackbar/ForceSnackbarTheme.kt:
```diff
@@ -0,0 +1,109 @@
+        key = "CornerRadius",
+        default = "8.0dip",
+        title = "Corner radius",
+        description = "Specify a corner radius for the snackbar."
```
```suggestion
        description = "The corner radius of the snackbar."
```

KobeW50 commented on src/main/kotlin/app/revanced/patches/youtube/general/snackbar/ForceSnackbarTheme.kt:
```diff
@@ -0,0 +1,109 @@
+            "Base" to "?ytBaseBackground"
+        ),
+        title = "Background color",
+        description = "Specify a background color for the snackbar. You can specify hex color."
```
```suggestion
        description = "The background color of the snackbar. Can be a hex color (#AARRGGBB) or a color resource reference."
```

(I don't actually know what a color resource reference, but I'm assuming that the stuff like `?ytBaseBackground` falls into that category. If not, remove that part from my suggestion.)

KobeW50 commented on src/main/kotlin/app/revanced/patches/youtube/general/snackbar/ForceSnackbarTheme.kt:
```diff
@@ -0,0 +1,109 @@
+            "Inverted" to "?attr/ytInvertedBackground"
+        ),
+        title = "Stroke color",
+        description = "Specify a stroke color for the snackbar. You can specify hex color."
```
```suggestion
        description = "The stroke color of the snackbar. Can be a hex color (#AARRGGBB) or a color resource reference."
```

(I don't actually know what a color resource reference, but I'm assuming that the stuff like `?attr/ytInvertedBackground` falls into that category. If not, remove that part from my suggestion.)

Also, wouldn't "border color" or "accent color" be better than "stroke color"?

KobeW50 commented on src/main/kotlin/app/revanced/patches/youtube/layout/translations/TranslationsPatch.kt:
```diff
@@ -1,41 +1,97 @@
 object TranslationsPatch : BaseResourcePatch(
     name = "Translations",
-    description = "Adds Crowdin translations for YouTube.",
+    description = "Add Crowdin translations for YouTube.",
```
```suggestion
    description = "Adds Crowdin translations for YouTube.",
```

KobeW50 commented on src/main/kotlin/app/revanced/patches/youtube/layout/splashanimation/AddSplashAnimationPatch.kt:
```diff
@@ -0,0 +1,59 @@
+@Suppress("DEPRECATION", "unused")
+object AddSplashAnimationPatch : ResourcePatch(
+    name = "Add splash animation",
+    description = "Adds old style splash animation.",
```
```suggestion
    description = "Adds, at compile time, the old style splash animation.",
```

KobeW50 commented on src/main/kotlin/app/revanced/patches/youtube/layout/translations/TranslationsPatch.kt:
```diff
@@ -1,41 +1,97 @@
+    private var SelectedLanguages by stringPatchOption(
+        key = "SelectedLanguages",
+        default = LANGUAGES.joinToString(", "),
+        title = "Selected RVX languages",
+        description = "Selected RVX languages that will be added."
```
```suggestion
        title = "RVX languages to add",
        description = "A list of languages to be added for the RVX settings, separated by commas."
```

KobeW50 replied:
@inotia00 Why was this only [commited](https://github.com/inotia00/revanced-patches/pull/55/commits/c9c3c90eff3ee82cb8a455f83bc2a1172c94414f) for YTM but not for YT?

(Same question for [this](https://github.com/inotia00/revanced-patches/pull/55#discussion_r1622994242), which had the YTM equivalent [committed](https://github.com/inotia00/revanced-patches/pull/55/commits/45efad2263cff01f5473d4be0767dcea67ac41a7))

inotia00 replied:
This suggestion has been reflected

KobeW50 commented on src/main/kotlin/app/revanced/patches/youtube/layout/translations/TranslationsPatch.kt:
```diff
@@ -1,41 +1,97 @@
+    private var SelectedAppLanguages by stringPatchOption(
+        key = "SelectedAppLanguages",
+        default = APP_LANGUAGES.joinToString(", "),
+        title = "Selected app languages",
+        description = "Selected app languages that will be kept, languages that are not in the list will be removed from the app."
```
```suggestion
        title = "App languages to keep",
        description = "A list of languages to be kept for the app, separated by commas. Languages not in the list will be removed from the app."
```

KobeW50 commented on src/main/kotlin/app/revanced/patches/youtube/layout/translations/TranslationsPatch.kt:
```diff
@@ -1,41 +1,97 @@
+        default = "",
+        title = "Custom language file",
+        description = """
+            The file path to the strings.xml file.
```
```suggestion
            The path to the strings.xml file.
```

Also, maybe `strings.xml` should have some sort of quotes around it. Such as like this:

```
The path to the 'strings.xml' file.
```

KobeW50 commented on src/main/kotlin/app/revanced/patches/youtube/shorts/overlaybuttons/ShortsOverlayButtonsPatch.kt:
```diff
@@ -0,0 +1,86 @@
+@Suppress("unused")
+object ShortsOverlayButtonsPatch : BaseResourcePatch(
+    name = "Shorts overlay buttons",
+    description = "Apply the new icons to the action buttons of the Shorts player.",
```
```suggestion
    description = "Applies the new icons to the action buttons in the Shorts player.",
```

KobeW50 commented on src/main/kotlin/app/revanced/patches/youtube/shorts/overlaybuttons/ShortsOverlayButtonsPatch.kt:
```diff
@@ -0,0 +1,86 @@
+            "OutlineCircle" to "outlinecircle",
+            DEFAULT_ICON_KEY to "tiktok"
+        ),
+        title = "Icon type of Shorts",
+        description = "Apply different icons for Shorts action buttons."
```
```suggestion
        title = "Shorts icon style ",
        description = "The style of the icons for the action buttons in the Shorts player."
```

KobeW50 commented on src/main/kotlin/app/revanced/patches/youtube/utils/settings/SettingsPatch.kt:
```diff
@@ -38,6 +42,49 @@ object SettingsPatch : BaseResourcePatch(
+        default = DEFAULT_ELEMENT,
+        values = SETTINGS_ELEMENTS_MAP,
+        title = "Insert position",
+        description = "Specify the setting name before which the RVX setting should be inserted."
```
```suggestion
        description = "The settings menu name that the RVX settings menu should be above."
```

(I may be misunderstanding this option. Please inform me if I am)

KobeW50 commented on src/main/kotlin/app/revanced/patches/youtube/utils/settings/SettingsPatch.kt:
```diff
@@ -38,6 +42,49 @@ object SettingsPatch : BaseResourcePatch(
+    private val CustomName by stringPatchOption(
+        key = "CustomName",
+        default = DEFAULT_NAME,
+        title = "Setting Name",
+        description = "Specify a custom name for the Extended preference."
```
```suggestion
        title = "Settings menu name",
        description = "The name of the RVX settings menu."
```

KobeW50 reviewed (commented):
Strings

inotia00 committed:
feat: clarify patch option descriptions

Co-authored-by: KobeW50 <84587632+KobeW50@users.noreply.github.com>

inotia00 committed:
feat: clarify patch option

Co-authored-by: KobeW50 <84587632+KobeW50@users.noreply.github.com>

inotia00 committed:
feat: clarify patch option

Co-authored-by: KobeW50 <84587632+KobeW50@users.noreply.github.com>

inotia00 committed:
feat: clarify patch option

Co-authored-by: KobeW50 <84587632+KobeW50@users.noreply.github.com>

inotia00 committed:
feat: clarify patch option

Co-authored-by: KobeW50 <84587632+KobeW50@users.noreply.github.com>

inotia00 committed:
feat: clarify patch option

Co-authored-by: KobeW50 <84587632+KobeW50@users.noreply.github.com>

inotia00 committed:
feat: clarify patch option

Co-authored-by: KobeW50 <84587632+KobeW50@users.noreply.github.com>

inotia00 committed:
feat: clarify patch option

Co-authored-by: KobeW50 <84587632+KobeW50@users.noreply.github.com>

inotia00 committed:
feat: clarify patch option

Co-authored-by: KobeW50 <84587632+KobeW50@users.noreply.github.com>

YT-Advanced commented on src/main/kotlin/app/revanced/patches/music/layout/branding/icon/CustomBrandingIconPatch.kt:
```diff
@@ -108,7 +119,84 @@ object CustomBrandingIconPatch : BaseResourcePatch(
+                        ).forEach { resourceGroup ->
+                            context.copyResources("$resourcePath/resource", resourceGroup)
+                        }
+                    } catch (e: Exception) {
```
Why do you use try catch in this case ?

anddea replied:
To avoid potential errors since some of them might not have some icons but it could be rewritten and I don't remember if it gave any error at all.

YT-Advanced replied:
I see, but what if the first group have error, for example. Then the excution for other groups will be skipped -> patch didn't worked as intended. You shoud rewrite it

anddea replied:
True, feel free to rewrite it when you have time. 

I made like this for YTM but in a different way for YT, I didn't have time and patience at the time but people asked for it. I actually wanted to improve it and later I forgot about it as it worked without errors, and now I just copied from my repo.

YT-Advanced replied:
Yea, but I don't have write access in inotia repos 🫠

inotia00 replied:
resolved in https://github.com/inotia00/revanced-patches/pull/55/commits/243ad46507fc5bedcea004a1e421f3d9461132d8

inotia00 committed:
chore: lint code

inotia00 committed:
fix(YouTube Music/Custom branding icon YouTube Music): add header and splash icon of Revancify

inotia00 commented on src/main/kotlin/app/revanced/patches/music/misc/translations/TranslationsPatch.kt:
```diff
@@ -1,46 +1,87 @@
+        key = "SelectedAppLanguages",
+        default = APP_LANGUAGES.joinToString(", "),
+        title = "App languages to keep",
+        description = "A list of languages to be kept for the app, separated by commas. Languages not in the list will be removed from the app."
```
This feature removes the default string resource for YouTube or YouTube Music
Is my understanding correct?

If so, wouldn't it make sense to separate it into different patches?

Because the description of the `Translations` patch is:
```
Adds Crowdin translations for YouTube Music
```

KobeW50 replied:
How about changing the patch description to something like: `Changes the available languages in the app.` or `Adds or changes the available languages in the app using Crowdin.`

inotia00 replied:
Oh yeah, I never thought there was such a simple way

anddea replied:
Maybe it's also worth noting in the description that English is the default language and will never be removed. Some users have encountered difficulties understanding this and kept `en-*` for English only (e.g., en-uk, en-au). However, to ensure only English (US) remains, the list should be empty.

These options for RVX and YT/YTM languages reduce the app size. It might be beneficial to mention this for less tech-savvy individuals, as I've observed people questioning the necessity of these options.

KobeW50 replied:
> Maybe it's also worth noting in the description that English is the default language and will never be removed. Some users have encountered difficulties understanding this and kept `en-*` for English only (e.g., en-uk, en-au). However, to ensure only English (US) remains, the list should be empty.

Is the default English one known as `en-us` or does that term not exist since it is the core strings.xml?

What I mean to ask is which of the following potential descriptions is correct:

1: `A list of languages to be kept for the app, separated by commas. Except for English, languages not in the list will be removed from the app.`

2: `A list of languages to be kept for the app, separated by commas. Except for en-us, languages not in the list will be removed from the app.`

anddea replied:
The first one is correct. Default English does not have language code. Even if they (users) delete all language codes, including all `en-*`, English will always be there.

KobeW50 committed:
refactor: Correct Translations patch description

MondayNitro commented:
this might not be very useful , but how about keeping the default order like revanced and adding a patch that allows user to specify his prefererred order for rvx settings in options.json? keeping general at top seems most sensible but who knows some users might like misc at top🙂

anddea commented:
> a patch that allows user to specify his prefererred order for rvx settings in options.json

It can also be done within settings, although I don't feel like doing it in the near future. And now, who will need it if there's a search option?

<details><summary>Unnecessary info</summary>
A year or two ago, I had implemented a sorting option within the RVX settings (kind of), but I made it locally and lost it when I merged the new code (year(s) ago). I don't remember if I ever opened a PR for this in any repo, but that was my intention. 
</details>

inotia00 commented on src/main/kotlin/app/revanced/patches/music/misc/translations/TranslationsPatch.kt:
```diff
@@ -1,46 +1,87 @@
     dependencies = setOf(SettingsPatch::class),
     compatiblePackages = COMPATIBLE_PACKAGE
 ) {
+    private var CustomLanguage by stringPatchOption(
```
The `Translations` (YouTube) patch and the `Translations` (YouTube Music) patch have patch options `CustomLanguage`, `SelectedLanguages`, and `SelectedAppLanguages` respectively.

Due to structural limitations of ReVanced Patcher, only one patch option is generated **if the patch name and patch option key are the same**.

To work around this structural limitation, patches `Translations` (YouTube) and `Translations` (YouTube Music) should have different patch names.

As a representative example, the `Custom branding icon` (YouTube) and `Custom branding icon` (YouTube Music) patches were changed to `Custom branding icon YouTube` and `Custom branding icon YouTube Music`, respectively.

inotia00 committed:
fix(Translations): Apply code review suggestions

inotia00 committed:
fix(Translations): Apply code review suggestions

inotia00 commented on src/main/kotlin/app/revanced/patches/youtube/shorts/overlaybuttons/ShortsOverlayButtonsPatch.kt:
```diff
@@ -0,0 +1,86 @@
+
+@Suppress("unused")
+object ShortsOverlayButtonsPatch : BaseResourcePatch(
+    name = "Shorts overlay buttons",
```
The `Overlay buttons` patch adds new player overlay buttons not provided by YouTube - such as Copy video URL and Whitelist.

The patch name `Shorts overlay buttons` can be confusing as it seems to be adding new Shorts overlay buttons that are not provided by YouTube.

The patch name should be changed to something like `Shorts action buttons theme`

ILoveOpenSourceApplications replied:
What about `Custom Shorts action buttons`?

anddea replied:
> What about `Custom Shorts action buttons`?

This one is good and consistent with similar patches. 

I named it like that because `Overlay buttons` also included Fullscreen button which already was there, but I'm fine with anything.

inotia00 commented on src/main/kotlin/app/revanced/patches/youtube/general/snackbar/ForceSnackbarTheme.kt:
```diff
@@ -0,0 +1,130 @@
+import org.w3c.dom.Element
+
+@Suppress("DEPRECATION", "unused")
+object ForceSnackbarTheme : BaseResourcePatch(
```
Since some snackbar themes do not seem to change, this patch will not be reflected in PR
(Theme of the snackbar should be changed programmatically)

If I find a way to change the theme of all snackbars in the future or find a way to apply the theme of snackbar programmatically, I will add it to the patch

inotia00 commented:
I'm debating whether or not to add the `Add splash animation` patch.
The reasons for this are as follows:

**1. As everyone knows, Google has rolled out the new Splash animation**
: [Google has widely rolled out Splash animation used on Android TV](https://www.rprna.com/apps/youtube-splash-screen-animation/)

Accordingly, Google has removed the old splash animation from YouTube v18.19.36.

The setting `Disable splash animation` in RVX only disables new splash animation.

If the user includes the `Add splash animation` patch, splash animation will not be disabled even if the `Disable splash animation` setting is enabled.

This is expected to cause user confusion, so I removed the `Add splash animation` patch when refactoring RVX patches. (revanced-patches-4.7.1)

**2. The MMT icon in `Custom branding icon YouTube` relies on the `Add splash animation` patch**
: To use custom splash animation, we need the `Add splash animation` patch.

If all icon presets support custom splash animation, this isn't a big problem.

Currently only the `MMT` icon supports custom splash animation.

In other words, the `Add splash animation` patch is required for custom splash animation of the `MMT` icon, which makes the code of the `Custom branding icon YouTube` patch overly complicated.

inotia00 commented:
The workaround I'm thinking about for this is:

**1. Simply remove the `Add splash animation` patch**
: This is the simplest workaround.

There is no need to refactor patches to improve code quality.

**2. Create custom splash animation for all custom branding icons**
: It is not an easy task.

The custom splash animation for the `MMT` icon did not take long to create, as it added minimal changes to the default splash animation.

If you try to create a custom splash animation for a custom branding icon from scratch, it will take a very long time.

<details>
    <summary>How to create custom splash animation</summary>

1. First, create a video to be used for splash animation. It doesn't matter whether it's a gif or mp4 format.

2. Split the frames of the video and convert them into image files. (20 to 80 image files are required.)

3. Convert all images to vector drawable.

4. Create an animated vector based on the time interval at which the vector drawable will be displayed.
(Loads 20 to 80 vector drawables in 2 to 5 seconds)

</details>

Then, supplement the description of the `Disable splash animation` setting.

MondayNitro commented:
> The workaround I'm thinking about for this is:
> 
> **1. Simply remove the `Add splash animation` patch** : This is the simplest workaround.
> 
> There is no need to refactor patches to improve code quality.
> 
> **2. Create custom splash animation for all custom branding icons** : It is not an easy task.
> 
> The custom splash animation for the `MMT` icon did not take long to create, as it added minimal changes to the default splash animation.
> 
> If you try to create a custom splash animation for a custom branding icon from scratch, it will take a very long time.
> 
> Then, supplement the description of the `Disable splash animation` setting.

please keep the add splash animation patch, i think it looked great (the new one is infact bad), you could rename the patches to something like "Add old splash animation" & "Disable new splash animation". the descriptions supplies alongside the patches will further clarify the patches' purpose to users

inotia00 commented:
> please keep the add splash animation patch

To maintain it, you'll need to refactor the patch
Current code is overly complex

Francesco146 commented:
what about explaining better the patch description, so the users know what happens when the patch is included. about adding the splash animation to the existing icons, should be a doable task, if i get the first one, then it's a mechanic thing to do. don't know if it's worth it, the code need to be refactored for sure

MondayNitro commented:
> what about explaining better the patch description, so the users know what happens when the patch is included. about adding the splash animation to the existing icons, should be a doable task, if i get the first one, then it's a mechanic thing to do. don't know if it's worth it, the code need to be refactored for sure

this might be useful [https://www.youtube.com/watch?v=Oy0oXwv3kSc](https://www.youtube.com/watch?v=Oy0oXwv3kSc) but svg icons will be needed

anddea commented:
Animations can be created by using online-based editors like Shape Shifter or more advanced software like After Effects and Unreal Engine. However, MMT animation was provided by its creator, kazimmt, and I believe the creators of those icons could provide their animations like true authors of those icons. 

Not sure if we should do it as they (the authors) can get upset.

inotia00 commented:
I will try to keep the old splash animation as much as possible

Perhaps the `Add splash animation` patch will be integrated into the `Custom branding icon YouTube` patch

inotia00 committed:
refactor(Custom branding icon): integrated `Add splash animation` patch, change patch options, change resource path

inotia00 committed:
feat(YouTube): remove `Force snackbar theme` patch https://github.com/inotia00/revanced-patches/pull/55#discussion_r1623865222

inotia00 committed:
refactor(Custom branding name): add the name of the vanilla app

inotia00 committed:
refactor(Settings): instead of using try..catch, remove all string elements

KobeW50 commented on src/main/resources/youtube/settings/host/values/strings.xml:
```diff
@@ -992,6 +993,9 @@ Limitation: Official headers in search results will be hidden."</string>
     <string name="revanced_hide_shorts_info_panel_summary_on">Info panels are hidden.</string>
     <string name="revanced_hide_shorts_info_panel_summary_off">Info panels are shown.</string>
+    <string name="revanced_hide_shorts_live_header_title">Hide live chat header</string>
+    <string name="revanced_hide_shorts_live_header_summary_on">Live chat header is hidden.\n\nBack button in header won\'t be hidden.</string>
+    <string name="revanced_hide_shorts_live_header_summary_off">Live chat header is shown.\n\nBack button in header won\'t be hidden.</string>
```
```suggestion
    <string name="revanced_hide_shorts_live_header_summary_on">Live chat header is hidden.\n\nBack button in header will not be hidden.</string>
    <string name="revanced_hide_shorts_live_header_summary_off">Live chat header is shown.\n\nBack button in header will not be hidden.</string>
```

inotia00 committed:
Update src/main/resources/youtube/settings/host/values/strings.xml

Co-authored-by: KobeW50 <84587632+KobeW50@users.noreply.github.com>

inotia00 committed:
refactor(YouTube/Settings): minimize changes from `dev` branch, add some missing strings, fix typo

inotia00 committed:
refactor(YouTube/Settings): minimize changes from `dev` branch, add some missing strings, fix typo

inotia00 commented on src/main/kotlin/app/revanced/patches/youtube/utils/settings/VisualPreferencesIconsPatch.kt:
```diff
@@ -0,0 +1,380 @@
+            ResourceGroup("drawable", *validTitlesIcons.values.map { "$it.xml" }.toTypedArray())
+        )
+
+        fun copyResourcesWithFallback(iconPath: String) {
```
This function is used in only one instruction
Unless the function is used in two or more instructions, there is no need to declare it as a function

inotia00 commented on src/main/kotlin/app/revanced/patches/youtube/utils/settings/VisualPreferencesIconsPatch.kt:
```diff
@@ -0,0 +1,380 @@
+            try {
+                context.copyResources(
+                    iconPath,
+                    ResourceGroup("drawable", "revanced_extended_settings_key_icon.xml")
```
`ResourceGroup("drawable", "revanced_extended_settings_key_icon.xml")` is used in two places, so you can declare it as a variable

inotia00 commented on src/main/kotlin/app/revanced/patches/youtube/utils/settings/VisualPreferencesIconsPatch.kt:
```diff
@@ -0,0 +1,380 @@
+        )
+
+        @Suppress("EXTENSION_SHADOWED_BY_MEMBER")
+        fun Document.getElementsByTagName(tagName: String): NodeList {
```
This function is not needed

inotia00 commented on src/main/kotlin/app/revanced/patches/youtube/utils/settings/VisualPreferencesIconsPatch.kt:
```diff
@@ -0,0 +1,380 @@
+        resourcesToCopy.forEach { context.copyResources("youtube/settings/icons", it) }
+
+        // Edit Preferences / add icon attribute
+        val tagNames = listOf(
```
Since `android:key` is a unique value, instead of using getElementsByTagName, you can use `Node.doRecursively`

inotia00 committed:
refactor(YouTube/Settings): Apply code review suggestions, Change to a more appropriate resource path, Clarify patch option, Excluded by default

inotia00 commented on src/main/kotlin/app/revanced/patches/youtube/shorts/overlaybuttons/ShortsOverlayButtonsPatch.kt:
```diff
@@ -0,0 +1,86 @@
+    compatiblePackages = COMPATIBLE_PACKAGE,
+    use = true
+) {
+    private const val DEFAULT_ICON_KEY = "TikTok"
```
As per [comment](https://github.com/inotia00/revanced-patches/pull/55#discussion_r1623165937
), the patch option `TikTok` needs to be renamed to a more appropriate name to reduce user confusion

`Round` or `Alternative` are the candidates I was thinking of

anddea replied:
I guess both are fine

inotia00 committed:
fix(YouTube/Settings): Restore `Closeable` interface, remove unused resources, add missing dependencies

inotia00 committed:
feat(YouTube): Rename `Shorts overlay buttons` to `Custom shorts action buttons`, change patch option name

inotia00 committed:
feat(YouTube): Change to more appropriate resource path

inotia00 committed:
fix typo

inotia00 committed:
feat: Reflect changes in the `dev` branch

inotia00 committed:
fix(YouTube/Shorts components): `Hide disabled comments button` setting is visible on YouTube versions prior to 18.33.40

inotia00 committed:
Merge branch 'dev' into killer

inotia00 commented:
Thanks

vippium commented:
In Visual Preference icons, as it is known that @inotia00 haven't included the "Extended name" and "extended icon" for options.json. It is good to see this. But the default icon for "Revanced Extended settings" is still set to "extension" which should be replaced with "Revanced" icon, to match it properly. As extension icon doesn't suits to Revanced settings.

So, pls change the "ExtendedIcon" from "Extension" to "Revanced" or something like that.

Here is the icon image: 
![Screenshot_2024_0611_064101](https://github.com/inotia00/revanced-patches/assets/93650504/43a5c1be-2afa-4605-a72e-99833c71b8f2)

anddea commented:
> the default icon for "Revanced Extended settings" is still set to "extension" which should be replaced with "Revanced" icon, to match it properly. As extension icon doesn't suits to Revanced settings.

"Extension" icon is used because these settings are Extensions for RV Extended, just like in the browser, you have normal settings and you can extend them with extensions, but you have plenty of options there if you want to change the dafault, including custom branding icons.

vippium commented:
> you have plenty of options there if you want to change the dafault.

No option...for the change, there should be extendedIcon key, which is not there. As I think, he removed it from options file

Git-Boy-Slim commented:
> No option...for the change, there should be extendedIcon key, which is not there. As I think, he removed it from options file

You have indeed an option in the options.json under "Visual preferences icons", where you can change the settings icon of the revanced settings ("key": "RVXSettingsMenuIcon"). The name ("key": "RVXSettingsMenuName") and position ("key": "InsertPosition") is changeable with the "Settings" option.

SaintNikku commented:
After this merge, "Revancify Yellow" can't be used with inotia00 patches whereas it was possible in anddea patches.

inotia00 commented:
> After this merge, "Revancify Yellow" can't be used with inotia00 patches whereas it was possible in anddea patches.

"Revancify Yellow" icon was not included in this PR
If you want, you can open a new PR

SaintNikku commented:
> > After this merge, "Revancify Yellow" can't be used with inotia00 patches whereas it was possible in anddea patches.
> 
> "Revancify Yellow" icon was not included in this PR If you want, you can open a new PR

@anddea
I have no idea what files are related to this yellow icon, can you kindly open a new PR to include that or guide me to correct commits and I'll do so.

Thank you.

anddea commented:
I'll open more PRs with new fixes and features when I have time and patience. 

There was no Revancify Yellow or new splash animations for AFN icons at the time when this PR was created.

Francesco146 commented:
I'll take care of the pr about Revancify yellow

weversonluizweversonluiz455-blip commented:
Dafault Music Player

weversonluizweversonluiz455-blip commented:
Dafault Music Player

weversonluizweversonluiz455-blip commented:
Dafault Music Player 
![Uploading images.jpeg…]()
